SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMovement/SpatialTestCharacterMovement.cpp

```diff
@@ -50,110 +50,108 @@ void ASpatialTestCharacterMovement::BeginPlay()
-			}
-			else
-			{
-				Test->AssertTrue(Test->bCharacterReachedDestination, TEXT("Player character has reached the destination on the autonomous proxy."));
+			PlayerCharacter->AddMovementInput(PlayerCharacter->GetActorForwardVector(), 1.0f);
 
-				Test->FinishStep();
+			if(bCharacterReachedDestination)
+			{
+				AssertTrue(bCharacterReachedDestination, TEXT("Player character has reached the destination on the autonomous proxy."));
+				FinishStep();
 			}
-		});
+		}, 1.0f);
 
 	// Server asserts that the character of client 1 has reached the Destination.
-	AddServerStep(TEXT("SpatialTestChracterMovementServerCheckMovementVisibility"), 1, nullptr, nullptr, [](ASpatialFunctionalTest* NetTest, float DeltaTime) {
-		ASpatialTestCharacterMovement* Test = Cast<ASpatialTestCharacterMovement>(NetTest);
-
-		Test->AssertTrue(Test->bCharacterReachedDestination, TEXT("Player character has reached the destination on the server."));
-
-		Test->FinishStep();
-		});
+	AddServerStep(TEXT("SpatialTestChracterMovementServerCheckMovementVisibility"), 1, nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+		{
+			if (bCharacterReachedDestination)
+			{
+				AssertTrue(bCharacterReachedDestination, TEXT("Player character has reached the destination on the server."));
+				FinishStep();
+			}
+		}, 1.0f);
 
 	// Client 2 asserts that the character of client 1 has reached the Destination.
-	AddClientStep(TEXT("SpatialTestCharacterMovementClient2CheckMovementVisibility"), 2, nullptr, [](ASpatialFunctionalTest* NetTest) {
-		ASpatialTestCharacterMovement* Test = Cast<ASpatialTestCharacterMovement>(NetTest);
-
-		Test->AssertTrue(Test->bCharacterReachedDestination, TEXT("Player character has reached the destination on the simulated proxy"));
-
-		Test->FinishStep();
-		});
-
+	AddClientStep(TEXT("SpatialTestCharacterMovementClient2CheckMovementVisibility"), 2, nullptr, nullptr, [this](ASpatialFunctionalTest* NetTest, float DeltaTime)
+		{
+			if (bCharacterReachedDestination)
+			{
+				AssertTrue(bCharacterReachedDestination, TEXT("Player character has reached the destination on the simulated proxy"));
+				FinishStep();
+			}
+		}, 1.0f);
 
 	// Universal clean-up step to delete the TriggerBox from all connected clients and servers		
-	AddUniversalStep(TEXT("SpatialTestCharacterMovementUniversalCleanUp"), nullptr, [](ASpatialFunctionalTest* NetTest) {
-		TArray<AActor*> FoundTriggers;
-		UGameplayStatics::GetAllActorsOfClass(NetTest->GetWorld(), ATriggerBox::StaticClass(), FoundTriggers);
-
-		for (auto TriggerToDestroy : FoundTriggers)
+	AddUniversalStep(TEXT("SpatialTestCharacterMovementUniversalCleanUp"), nullptr, [this](ASpatialFunctionalTest* NetTest)
 		{
-			TriggerToDestroy->Destroy();
-		}
+			TArray<AActor*> FoundTriggers;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), ATriggerBox::StaticClass(), FoundTriggers);
+
+			for (auto TriggerToDestroy : FoundTriggers)
+			{
+				TriggerToDestroy->Destroy();
```

Comment:
Could also auto-register

Reply:
Ah, I now realize why the first step is universal. The triggers are created locally for both the server and the client and they are not replicated. I believe AutoRegisterDestroyActor *could* still work, but needs a try.

Reply:
Yes, I have tried that but it requires the Test to have an owning connection, that as far as I understand would then require the server to set the owner of the Test in another step, might be better to have the trigger created by the server and have it replicate I think, since it shouldn't change the scenario that I am testing.

Reply:
Having them replicate is okay, I think.

Reply:
TL;DR; After trying, I think removing the universal step and making the trigger replicate is not providing much value.

Adding that then requires the `bCharacterReachedDestination` to be replicated since the collision delegate will then only be executed on the server,, and also resetting it requires a Server RPC, making the test code harder to maintain IMO. Let me know if you think the ServerStep would still be better.

Reply:
Note that after Victoria's PR on RegisterAutoDestroy, we should be able to register client non-replicated actors as well.

Reply:
Shall I wait for that PR to merge, pull it and register the TriggerBox to auto destroy and then update this PR?

Reply:
Not that crucial, whichever goes first, goes first.